torch/csrc/jit/codegen/cuda/transform_replay.cpp

```diff
@@ -430,7 +430,7 @@ std::pair<TensorDomain*, unsigned int> TransformReplay::replayCasP(
     size_t itc = 0, itp = 0;
     while (itc < consumer_root.size() || itp < producer_root.size()) {
       if (itc < consumer_root.size() && consumer_root[itc]->isBroadcast() &&
-          (itp > producer_root.size() || !producer_root[itp]->isBroadcast())) {
+          (itp >= producer_root.size() || !producer_root[itp]->isBroadcast())) {
```

Comment:
Is this related to `cache_before/after`? Or, does this fix an existing bug?

Reply:
This is a bug I found while working on cache_before/after. If `itp == producer_root.size()`, then `producer_root[itp]` goes out of bounds.